python/cudf/cudf/core/column/numerical.py

```diff
@@ -453,7 +453,10 @@ def _safe_cast_to_int(col, dtype):
 
 
 def _normalize_find_and_replace_input(input_column_dtype, col_to_normalize):
-    normalized_column = column.as_column(col_to_normalize)
+    normalized_column = column.as_column(
+        col_to_normalize,
+        dtype=input_column_dtype if len(col_to_normalize) <= 0 else None,
```

Comment:
why default the dtype to float64 here?

Reply:
So here we are actually trying to create a `column` out of `col_to_normalize`, but whenever `col_to_normalize` is empty we still default to `float` dtype, now with this change we want to assign the input column dtype(`input_column_dtype`) to `normalized_column`